### PR TITLE
Cursor

### PR DESCRIPTION
**Pull request by norkans7**

### CODE REVIEW

File: temba_client/clients.py [hunk 1]
```diff
@@ -3,6 +3,14 @@
 import datetime
 import json
 import logging
+
+try:
```
Comment (rowanseymour) [inline temba_client/clients.py]: Not sure but I think you can do `from six.moves.urllib.parse import ...`

**Comment by rowanseymour** [comment]
I think there's a simpler approach we can take here that avoids adding a cursor param to every fetch method. Most of the fetch methods of the client return a `CursorQuery` which has  an `iterfetches` method which returns a `CursorIterator`. Iterators are already resumable - for example this is possible with no changes to the code:

``` python
contact_iterator = client.get_contacts(after=dt).iterfetches()
for contact_batch in contact_iterator:
  if num_contacts > 100000:
     break;
  ...

for contact_batch in contact_iterator:
  ... # resumes where we left off
```

The only limitation with the above approach is that `contact_iterator` is a Python object and not something you can easily stuff in Redis etc. If that's something you need to do then we could make it serializable to JSON or something like that.

Or we make it so that you can extract the cursor value (a string which can be easily persisted somewhere) from the iterator (like you've done in this PR), but let it passed back into `iterfetches` - for example:

``` python
contact_iterator = client.get_contacts(after=dt).iterfetches()

for contact_batch in contact_iterator:
  if num_contacts > 100000:
     cursor = contact_iterator.get_cursor()
     break;
  ...

contact_iterator = client.get_contacts(after=dt).iterfetches(resume_cursor=cursor)
for contact_batch in contact_iterator:
   ...
```

If serializing the iterator is something you need to do then I'd prefer the latter approach. If not then I don't think anything needs to change.

**Comment by norkans7** [comment]
Actually we need to break and let another poll fetch the runs, I think we'll be returning from the method and just saving the latest cursor to use later on another client from a new task.

**Comment by rowanseymour** [comment]
So `client.get_contacts(after=dt).iterfetches(resume_cursor=cursor)` should work for you right?

**Comment by norkans7** [comment]
Ohh so you mean to pass the cursor to iterfetches instead of the client methods, now I get that,

I think that could work

**Comment by norkans7** [comment]
@rowanseymour please review once again

**Comment by rowanseymour** [comment]
👍

**Pull request merged**